### PR TITLE
Власов М. С. 381806-1 ЛР3

### PR DESCRIPTION
**Pull request by vla5924**
### Pull Request Readiness Checklist

See details at [code review]( https://github.com/UNN-ITMM-Software/devtools-course-practice/blob/main/lab-guide/lab-1-setup.md#%D1%80%D0%B5%D1%86%D0%B5%D0%BD%D0%B7%D0%B8%D1%8F-%D0%BA%D0%BE%D0%B4%D0%B0)

- [ ] Code is well documented and more information can be seen on the [website](https://devtools-course-practice-2021.readthedocs.io/en/latest/modules.html) (recommended)
- [ ] Make sure that the testing has passed successfully on [GitHub Actions](https://github.com/UNN-ITMM-Software/devtools-course-practice/actions)
- [ ] Get two approve from your fellow students
- [ ] Take a review from one of the administrators

### CODE REVIEW

File: modules/qsort/src/qsorter.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                   std::string(e.what());
+        } catch (std::out_of_range& e) {
+            return "[ERROR] " + arg + ": out of range. " +
+                   std::string(e.what());
+        }
+    }
+    try {
+        Sort(&numbers, 0, numbers.size() - 1);
+    } catch (const char* e) {
+        return "[ERROR] Unable to sort. " + std::string(e);
```
Comment (vlad-gogov) [inline modules/qsort/src/qsorter.cpp:37]: Лучше ловить и бросать std::exception

**Review by vlad-gogov — CHANGES_REQUESTED** [review]

File: modules/qsort/src/qsorter.cpp (hunk 1 above)
Comment (vla5924) [reply modules/qsort/src/qsorter.cpp]: Fixed

**Review by vlad-gogov — APPROVED** [review]

File: modules/qsort/src/qsorter.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        }
+    }
+    try {
+        Sort(&numbers, 0, numbers.size() - 1);
+    } catch (std::runtime_error& e) {
+        return "[ERROR] Unable to sort. " + std::string(e.what());
+    }
+    std::string result = std::to_string(numbers[0]);
+    for (size_t i = 1; i < numbers.size(); i++)
+        result += " " + std::to_string(numbers[i]);
```
Comment (MaksimShagov) [inline modules/qsort/src/qsorter.cpp:41]: Можно использовать stringstream

**Review by MaksimShagov — CHANGES_REQUESTED** [review]

File: modules/qsort/src/qsorter.cpp (hunk 2 above)
Comment (vla5924) [reply modules/qsort/src/qsorter.cpp]: Fixed

**Review by MaksimShagov — APPROVED** [review]

**Review by dkurt — APPROVED** [review]

**Pull request merged**